File: finance-app-backend/src/main/java/com/victor/financeapp/backend/application/dto/TransactionDTO.java
package com.victor.financeapp.backend.application.dto;

import com.victor.financeapp.backend.domain.model.common.Transaction.TransactionType;
import lombok.Builder;

import com.victor.financeapp.backend.domain.model.common.Category;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record TransactionDTO(
        Long id,
        String description,
        BigDecimal amount,
        String type,
        LocalDateTime date,
        Category category,
        boolean isInstallment,
        BigDecimal installmentAmount,
        String installmentId,
        int installmentNumber,
        Long creditCardId
) {
}

File: finance-app-android/app/src/main/java/com/viictrp/financeapp/data/remote/dto/TransactionDTO.kt
package com.viictrp.financeapp.data.remote.dto

import androidx.compose.runtime.Immutable
import com.viictrp.financeapp.domain.model.transaction.TransactionType
import java.math.BigDecimal
import java.time.LocalDateTime

@Immutable
data class TransactionDTO(
    val id: Long? = null,
    val description: String,
    val amount: BigDecimal,
    val category: String,
    val type: TransactionType,
    val date: LocalDateTime,
    val isInstallment: Boolean? = false,
    val installmentAmount: Int? = 1,
    val installmentId: String? = null,
    val installmentNumber: Int? = 1,
    val creditCardId: Long?,
)